sktree/tests/test_honest_forest.py

```diff
@@ -0,0 +1,139 @@
+import time
+
+import numpy as np
+import pytest
+from sklearn_fork import datasets
+from sklearn_fork.metrics import accuracy_score
+
+from sktree.ensemble import HonestForestClassifier
+
+CLF_CRITERIONS = ("gini", "entropy")
+
+# also load the iris dataset
+# and randomly permute it
+iris = datasets.load_iris()
+rng = np.random.RandomState(1)
+perm = rng.permutation(iris.target.size)
+iris.data = iris.data[perm]
+iris.target = iris.target[perm]
+
+
+def test_toy_accuracy():
+    clf = HonestForestClassifier(n_estimators=10)
+    X = np.ones((20, 4))
+    X[10:] *= -1
+    y = [0] * 10 + [1] * 10
+    clf = clf.fit(X, y)
+    np.testing.assert_array_equal(clf.predict(X), y)
+
+
+@pytest.mark.parametrize("criterion", ["gini", "entropy"])
+@pytest.mark.parametrize("max_features", [None, 2])
+def test_iris(criterion, max_features):
+    # Check consistency on dataset iris.
```

Comment:
Can we test various trees as input to `HonestForestClassifier`? 

E.g.

HonestTreeClassifier(ObliqueDecisionTree)
HonestTreeClassifier(PatchObliqueDecisionTree)